frontend/app/services/SubscriptionService.scala

```diff
@@ -132,4 +144,41 @@ class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String], val
       invoiceItems <- zuora.query[InvoiceItem](s"SubscriptionId='${subscription.current}'")
     } yield PaymentSummary(invoiceItems)
   }
+
+  def getMembershipSubscriptionSummary(memberId: MemberId): Future[MembershipSummary] = {
+
+    def futureFreeStartingPeriodOffer(memberId: MemberId) =
+      for (subscription <- getLatestSubscription(memberId))
```

Comment:
It looks like `getLatestSubscription(memberId)` gets called more than once in this method (here, and in `getPaymentSummaryForMembershipsWithFreePeriod()`), which is something to watch out for - it calls `getSubscriptions()`, which in turn hits the Zuora API: so performing superfluous duplicate calls to Zuora.